scipy/optimize/_differentialevolution.py

```diff
@@ -580,21 +592,29 @@ def __init__(self, func, bounds, args=(),
         else:
             self.init_population_array(init)
 
-        # infrastructure for constraints
-        # dummy parameter vector for preparing constraints, this is required so
-        # that the number of constraints is known.
-        x0 = self._scale_parameters(self.population[0])
+        if x0 is not None:
+            # scale to within unit interval and
+            # ensure parameters are within bounds.
+            x0_scaled = np.clip(
+                self._unscale_parameters(np.asarray(x0)), 0.0, 1.0
```

Comment:
Shouldn't `x0` ouside of specified `bounds` just raise a `ValueError`? Both are user-provided, so I'd consider the inconsistency to be user error. We typically prefer raising rather than silently changing the value.

Reply:
Also, can you add a test? There's still no test with `x0` being passed to the function that's public API.

Reply:
> Shouldn't x0 ouside of specified bounds just raise a ValueError? 

If the user supplies an initial population to `init` the existing behaviour is to [do the same thing](https://github.com/scipy/scipy/blob/master/scipy/optimize/_differentialevolution.py#L678) as this PR. I think it definitely makes sense to check `x0` behaviour for the deterministic minimizers, but it's less important here. The solver success isn't degraded by a specific `x0`, so is a check worthwhile adding?

As the function exists it's quite fault tolerant, making the check is going to make it more likely to raise Errors that the user has to deal with, and even if they do fix the error I don't think the eventual solve trajectory will change all that much.